### PR TITLE
python3Packages.spacy: 3.2.1 fix

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
the initial Idea was to fix `spacy`. But all fixes took more work then expected. The update was done by @jonringer.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 153771` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python38Packages.spacy</li>
    <li>python38Packages.spacy-loggers</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.textacy</li>
    <li>python38Packages.wandb</li>
    <li>python38Packages.yaspin</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.spacy-loggers</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.textacy</li>
    <li>python39Packages.wandb</li>
    <li>python39Packages.yaspin</li>
  </ul>
</details>

**Comment by risicle** [comment]
`nixpkgs-review` successful, macos 10.15 & nixos x86_64

**Comment by risicle** [comment]
`nixpkgs-review` successful, macos 10.15 & nixos x86_64

File: pkgs/development/python-modules/spacy-loggers/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ];
+
+  pythonImportsCheck = [ "spacy_loggers" ];
+
+  # skipping the checks, becaus it requires a cycle dependency to spacy as well.
+  doCheck = false;
+
+  checkInputs = [
+    pytest
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/spacy-loggers/default.nix:31]: Unittests are not executed.

```suggestion
```

File: pkgs/development/python-modules/yaspin/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+, callPackage
+, fetchPypi
+, buildPythonPackage
+, termcolor
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "yaspin";
+  version = "2.1.0";
```
Comment (fabaff) [inline pkgs/development/python-modules/yaspin/default.nix:11]: Minimal support Python release is [3.6](https://github.com/pavdmyt/yaspin/blob/master/pyproject.toml#L50).

```suggestion
  version = "2.1.0";

    disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/yaspin/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    inherit pname version;
+    sha256 = "sha256-yNNOyp/aP0375Z9X888PNkGvPu+/FUT765s7rPgsWAo=";
+  };
+
+  propagatedBuildInputs = [
+    termcolor
+  ];
+
+  checkInputs = [
+    pytest
```
Comment (fabaff) [inline pkgs/development/python-modules/yaspin/default.nix:23]: Can you please check if we could use `pytestCheckHook`?

**Review by fabaff — COMMENTED** [review]
New packages should have a maintainer set.

**Comment by sikmir** [comment]
Bump spacy to [3.2.1](https://github.com/explosion/spaCy/releases/tag/v3.2.1) please.

**Comment by Stunkymonkey** [comment]
because of #138474 i removed the duplicate commits adding `yaspin`&`wandb`.

**Comment by Stunkymonkey** [comment]
@sikmir & @jonringer anything to add?

**Review by sikmir — APPROVED** [review]

**Comment by sikmir** [comment]
`spacy` broken again on master caused by `thinc`.

**Pull request merged**